go/mysql/bufio_pool.go

```diff
@@ -0,0 +1,65 @@
+package mysql
+
+import (
+	"bufio"
+	"io"
+	"sync"
+)
+
+// Writer implementation
```

Comment:
Can you explain the idea here? How the buffered ios actually lifecycle, and how it's used in conn.go/query.go?

In particular, we always expect flush() to be called eventually after every protocol write, so we get good amortized behavior.